src/gui/plugins/scene3d/Scene3D.cc

```diff
@@ -1581,12 +1582,23 @@ void IgnRenderer::HandleMouseViewControl()
   }
   else
   {
-    if (this->dataPtr->mouseEvent.Type() == common::MouseEvent::PRESS)
+    if (this->dataPtr->mouseEvent.Type() == common::MouseEvent::PRESS ||
+        // the rendering thread may miss the press event due to
+        // race condition when doing a drag operation (press and move, where
+        // the move event overrides the press event before it is processed)
+        // so we double check to see if target is set or not
+        (this->dataPtr->mouseEvent.Type() == common::MouseEvent::MOVE &&
+        std::isinf(this->dataPtr->target.X())))
```

Comment:
Is it possible to get here while hovering instead of dragging?

Reply:
currently we only process drag events so it should not get here when hovering:
https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo3/src/gui/plugins/scene3d/Scene3D.cc#L3142-L3143

I've also added an extra check here for `Dragging()` so the logic here does not rely on other parts of scene3D.